MSMAll/scripts/SingleSubjectConcat.sh

```diff
@@ -31,504 +31,182 @@
-			VarVN="-var VN ${OutputVN} -select 1 1 -repeat"
-		else
-			MATHVN=""
-			VarVN=""
-		fi
-      
-		MATH="((TCS${MATHDemean})${MATHRB})${MATHVN}"
-		log_Msg "MATH: ${MATH}"
-    
-		${Caret7_Command} -cifti-math "${MATH}" ${ResultsFolder}/${fMRIName}${fMRIProcSTRING}${OutputProcSTRING}.dtseries.nii -var TCS ${ResultsFolder}/${fMRIName}${fMRIProcSTRING}.dtseries.nii ${VarDemean} ${VarRB} ${VarVN} 
-
-		if [ "${Demean}" = "YES" ] ; then
-			rm ${ResultsFolder}/${fMRIName}${fMRIProcSTRING}_mean.dscalar.nii
-		fi
-
-		if [ "${VarianceNormalization}" = "YES" ] ; then
-			if [ "${ComputeVarianceNormalization}" = "YES" ] ; then
-				rm ${ResultsFolder}/${fMRIName}${fMRIProcSTRING}_vn_tempcompute.dscalar.nii
-			fi
-		fi
-
-		MergeSTRING=`echo "${MergeSTRING} -cifti ${ResultsFolder}/${fMRIName}${fMRIProcSTRING}${OutputProcSTRING}.dtseries.nii"`
-		
-	done
-	mkdir -p "${OutputFolder}"
-	${Caret7_Command} -cifti-merge ${OutputFolder}/${OutputfMRIName}${fMRIProcSTRING}${OutputProcSTRING}.dtseries.nii ${MergeSTRING}
-}
-
-# ------------------------------------------------------------------------------
-#  "Global" processing - everything above here should be in a function
-# ------------------------------------------------------------------------------
-
-# Establish defaults
-G_DEFAULT_MATLAB_RUN_MODE=1		# Use interpreted MATLAB
-
-# Set global variables
-g_script_name=$(basename "${0}")
-
-# Allow script to return a Usage statement, before any other output
-if [ "$#" = "0" ]; then
-    show_usage
-    exit 1
-fi
-
-# Verify that HCPPIPEDIR environment variable is set
-if [ -z "${HCPPIPEDIR}" ]; then
-	echo "${g_script_name}: ABORTING: HCPPIPEDIR environment variable must be set"
-	exit 1
-fi
-
-# Load function libraries
-source "${HCPPIPEDIR}/global/scripts/debug.shlib" "$@"         # Debugging functions; also sources log.shlib
-source ${HCPPIPEDIR}/global/scripts/opts.shlib                 # Command line option functions
-
-opts_ShowVersionIfRequested $@
-
-if opts_CheckForHelpRequest $@; then
-	show_usage
-	exit 0
+  	# math expression
+	MATH="(TCS${MATHDemean})${MATHVN}"
+	log_Msg "MATH: ${MATH}"
+	
+	# demean + vn
+	${Caret7_Command} -cifti-math "${MATH}" ${ResultsFolder}/${fMRIName}${fMRIProcSTRING}${FrameString}${OutputProcSTRING}.dtseries.nii -var TCS ${ResultsFolder}/${fMRIName}${fMRIProcSTRING}${FrameString}.dtseries.nii ${VarDemean} ${VarVN} 
+	
+	# remove the mean files & dense timeseries with selected frame range to save space
+	rm ${ResultsFolder}/${fMRIName}${fMRIProcSTRING}${FrameString}_mean.dscalar.nii
+	rm ${ResultsFolder}/${fMRIName}${fMRIProcSTRING}${FrameString}.dtseries.nii
+	
+	# construct the merge string
+	MergeSTRING=`echo "${MergeSTRING} -cifti ${ResultsFolder}/${fMRIName}${fMRIProcSTRING}${FrameString}${OutputProcSTRING}.dtseries.nii"`
+	
+	# construct the remove string
+	RemoveSTRING=`echo "${RemoveSTRING} ${ResultsFolder}/${fMRIName}${fMRIProcSTRING}${FrameString}${OutputProcSTRING}.dtseries.nii"`
```

Comment:
If these always get deleted, consider `tempfiles.shlib`.

The backtick echo construct is not needed, `RemoveSTRING="$RemoveSTRING <whatever>"` works (and += can be used), but for whitespace safety (and not needing to remember the space character) arrays are a better approach: `RemoveARRAY+=("<whatever>")`, with a `RemoveARRAY=()` at the top and `rm -f "${RemoveARRAY[@]}"` at the bottom.  `tempfiles.shlib` does this all for you with just a `tempfiles_add "<whatever>"` command and no setup or `rm` at the end, and you can also instead use `tempfiles_create somename_XXXXXX.dtseries.nii mytempname` (the variable `mytempname` contains the actual filename to use) to create a file in the temporary directory, which can often do faster IO than writing to the subject dir.

Reply:
I added the `tempfiles_add`. This is a really cool function!